From 10f3baebdd237e2236b3bb653218ab7c5b623ccd Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Tue, 01 Apr 2025 21:39:13 +0300
Subject: [PATCH] [SW-224098] gaudi2: security: default bis for hl288

Since supermicro servers do not refresh PCI rev on host reboot, there is
a problem with using fake security/bis as rev stays 1 instead of 4.

In order to solve this, we set BIS (and rev 4) by default for HL-288
card (as it is a China only card).
This does not apply to fake bis/security so we may still have the
flexibly to test different scenarios.

Change-Id: Idcd3d3bdb8dc1785d3f8311276f4d1fe5208c530
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 6fc87ea..4a10ea3 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -360,16 +360,25 @@
 	 * To avoid potential mismatch (cold reset done with fake sec enabled)
 	 * fake fuse contents are not read from SPI if fake sec is not enabled
 	 */
-	if (preboot_fake_security_is_enabled() && preboot_hl_sec_fuse_cache.is_invalid) {
+	if (preboot_hl_sec_fuse_cache.is_invalid) {
 		fuse = &preboot_hl_sec_fuse_cache.fuse;
-		/* Copy efuse security data */
-		spi_flash_copy(&fuse->data, FAKE_SECURITY_FUSE_OFFSET + (EFUSE_SEC_OFFSET *
-				sizeof(*val)), sizeof(*fuse), "FAKE SECURITY FUSE");
 
-		/* Fake fuse data is stored in negative logic */
-		for (i = 0; i < ARRAY_SIZE(fuse->data); i++) {
-			val = &fuse->data[i];
-			*val = ~(*val);
+		if (preboot_fake_security_is_enabled()) {
+			/* Copy efuse security data */
+			spi_flash_copy(&fuse->data, FAKE_SECURITY_FUSE_OFFSET + (EFUSE_SEC_OFFSET *
+						sizeof(*val)), sizeof(*fuse), "FAKE SECURITY FUSE");
+
+			/* Fake fuse data is stored in negative logic */
+			for (i = 0; i < ARRAY_SIZE(fuse->data); i++) {
+				val = &fuse->data[i];
+				*val = ~(*val);
+			}
+			/* HL-288 is China only card so BIS D is set by default,
+			 * unless fake security is enabled
+			 */
+		} else if (is_card_type_pcie()) {
+			fuse->bis_compliant = 1;
+			fuse->bis_d_compliant = 1;
 		}
 
 		preboot_hl_sec_fuse_cache.is_invalid = false;
